redisinsight/api/src/modules/rdi/rdi-pipeline.service.spec.ts

```diff
@@ -0,0 +1,403 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { RdiClientProvider } from 'src/modules/rdi/providers/rdi.client.provider';
+import { RdiPipelineAnalytics } from 'src/modules/rdi/rdi-pipeline.analytics';
+import { wrapHttpError } from 'src/common/utils';
+import {
+  MockType, generateMockRdiClient, mockRdiClientProvider, mockRdiDryRunJob,
+} from 'src/__mocks__';
+import { RdiPipelineService } from './rdi-pipeline.service';
+import { RdiDryRunJobDto } from './dto';
+import { RdiDyRunJobStatus, RdiPipeline } from './models';
+
+describe('RdiPipelineService', () => {
+  let service: RdiPipelineService;
+  let rdiClientProvider: MockType<RdiClientProvider>;
+  let analytics: RdiPipelineAnalytics;
+  const rdiClientMetadata = { id: '123', sessionMetadata: undefined };
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        RdiPipelineService,
+        {
+          provide: RdiClientProvider,
+          useFactory: mockRdiClientProvider,
+        },
+        {
+          provide: RdiPipelineAnalytics,
+          useValue: {
```

Comment:
could you please move this to __mocks__ the same way as here `useFactory: mockRdiClientProvider,`?